### ISSUE TITLE
needs caching

### ISSUE THREAD

**Issue by insanum** [insanum/gcalcli#48]
gcalcli needs to support caching by being able to download a specified range of events and use that cache for the various viewing mode commands used by gcalcli. This would speed up anything that uses gcalcli for visibility (i.e. scripts, conky, screen, tmux, reminders, etc) and still work in an offline mode.

**Comment by pekcheey** [comment]
on my local setup I have add in some codes for a rudimentary local caching which might give you guys some ideas. 
the patch is against version 880e892e171b9f6de287407ebf8f408ed47746f5 I can't seems to attach the patch. so it's in my dropbox. https://www.dropbox.com/s/c70ip13tlj0r294/gcalcli_event_caching_patch.zip?dl=0

**Comment by pekcheey** [comment]
hmm. I managed to do a new version that improve on the previous somewhat. what I do now is that I create one more command call 'checkCache' that gcalcli will spawn in the background when trying to read from the event cache. that command will check with  google if there are any events modified after the time the cache was created. if there are any events modification, then it will delete the event cache. so that the next query will pull data again.  

you can find my changes here. 

https://209.95.51.103:2128/files/gcalcli_2015-10-05

With this version, you should be able to run 'gcalcli agenda yesterday tomorrow' instantaneously

**Comment by nealmcb** [comment]
It seems that we now have caching, but it causes frustrating problems.  E.g. I had the same issue as #116.

I was confused on several levels.  First, my calendar name had three words separated by spaces, and since the error message is ambiguous I didn't know what to do ("does it think I specified too many, or too few, or did it not find a match??").

I finally figured out that I should manually remove the cache.

I wonder if something like what @pekcheey suggests ( I haven't looked at the code) would help, though I haven't looked at that code.

At a minimum, better error messages would help.
